[quant][graphmode] Remove dedup pass

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39919 [wip][quant] Support general op modules with inplace options
* #39915 [quant][graphmode] Pass debug option into insert_quant_dequant pass
* #39911 [quant][graphmode] Use quantized::batch_norm in graph mode
* #39910 [quant] add quantized::batch_norm
* #39895 [quant][graphmode] Support prim::TupleUnpack and prim::TupleConstruct
* #39892 [quant][graphmode][refactor] isScalar check
* #39871 [quant][graphmode] Use list of filters in QuantFusionInfo
* #39867 [jit][subgraph_rewriter] Support list of filters
* #39826 [quant][graphmode] Quantizing traced modules
* **#39825 [quant][graphmode] Remove dedup pass**

Summary:
Removing the pass for now since it is causing error for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21987878](https://our.internmc.facebook.com/intern/diff/D21987878)

jerryzh168 committed:
[quant][graphmode] Remove dedup pass

Summary:
Removing the pass for now since it is causing error for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Remove dedup pass"


Summary:
Removing the pass for now since it is causing error for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Remove dedup pass"


Summary:
Removing the pass for now since it is causing error for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21987878](https://our.internmc.facebook.com/intern/diff/D21987878)

[ghstack-poisoned]

raghuramank100 commented on torch/quantization/quantize_script.py:
```diff
@@ -35,7 +35,6 @@ def _prepare_script(model, qconfig_dict, inplace=False, quant_type=QuantType.STA
     if not all(isinstance(x, str) for x in qconfig_dict.keys()):
         raise ValueError('qconfig_dict should only contain names(str) as keys.')
     scripted_qconfig_dict = script_qconfig_dict(qconfig_dict)
-    torch._C._jit_pass_dedup_module_uses(model._c)
```
Dont you need to remove the corresponding tests?

jerryzh168 replied:
this is a legacy pass we added for this case, but this case can be addressed with current code already. I feel this pass might not be needed if we have some (unlikely) case like:

```python
x = self.submodule_with_states(x)
...
x = self.submodule_with_states(x)
```

that is: we use some submodule with states multiple times.

I only removed this from API, this pass is still there. If we decide to support the above use case at some point, we'll debug the problem and re-enable the pass.

jerryzh168 committed:
Update on "[quant][graphmode] Remove dedup pass"


Summary:
Removing the pass for now since it is causing error for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21987878](https://our.internmc.facebook.com/intern/diff/D21987878)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Remove dedup pass"


Summary:
Removing the pass for now since it is causing error for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21987878](https://our.internmc.facebook.com/intern/diff/D21987878)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Remove dedup pass"


Summary:
Removing the pass for now since it is causing error for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21987878](https://our.internmc.facebook.com/intern/diff/D21987878)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Remove dedup pass"


Summary:
Removing the pass for now since it is causing error for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21987878](https://our.internmc.facebook.com/intern/diff/D21987878)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Remove dedup pass"


Summary:
Removing the pass for now since it is causing error for some models

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21987878](https://our.internmc.facebook.com/intern/diff/D21987878)

[ghstack-poisoned]
